doc/vxlan/EVPN/EVPN_VXLAN_HLD.md

```diff
@@ -0,0 +1,1434 @@
+
+```
+
+**NEIGH_SUPPRESS_APP_VLAN_TABLE**
+
+Producer: VlanMgr
+
+Consumer: VlanOrch
+
+Description: New table for per VLAN neighbor suppression configuration.
+
+Schema:
+
+```
+; New table
+;Stores Neigh Suppress configuration per VLAN
+
+key             = SUPPRESS_VLAN_NEIGH_TABLE:"Vlan"vlanid		; VLAN with prefix "NEIGH_SUPPRESS"
+suppress_neigh  = "ON" / "OFF" ; Default "OFF"
+```
+
+### 4.2.3 STATE_DB changes
+
+**STATE_VXLAN_TUNNEL_TABLE**
+
+Producer: vxlanorch
+
+Consumer: show scripts
+
+Description: 
+
+- Holds the created tunnels - static or dynamic. 
+- There is a VXLAN Tunnel entry for every tunnel created. 
+- The SIP and DIP are added for tunnels. 
+- Tunnel Creation source is added. 
+
+Schema:
+
+```
+; New table
+; Defines schema for VXLAN State
+key             = VXLAN_TUNNEL:name    ; tunnel name
+; field         = value
+SRC_IP          = ipv4                 ; tunnel sip
+DST_IP          = ipv4                 ; tunnel dip 
+tnl_src         = "CLI"/"EVPN"  
+operstatus      = "oper_up"/"oper_down"  
+```
+
+### 4.2.4 COUNTER_DB changes
+
+Producer: syncd
+
+Consumer: show scripts
+
+Description:
+
+- Per Tunnel Rx/Tx packets and octet counters are added.
+- Existing counters table is used to add the tunnel counters.
+
+Schema:
+
+```
+; Existing table
+key                           = COUNTERS:tunnel_vid   ; tunnel oid
+; field                       = value
+SAI_TUNNEL_STAT_IN_OCTETS     = number    ;uint64 
+SAI_TUNNEL_STAT_IN_PACKETS    = number    ;uint32  
+SAI_TUNNEL_STAT_OUT_OCTETS    = number    ;uint64 
+SAI_TUNNEL_STAT_OUT_PACKETS   = number    ;uint32
+```
+
+
+
+## 4.3 Modules Design and Flows
+
+#### EVPN Control Plane and configurations
+
+FRR as EVPN control plane is assumed in this document. However, any BGP EVPN implementation can be used as control plane as long as it complies with the EVPN design proposed in this document.
+
+Following configurations are common for enabling EVPN L2/L3 services:
+
+- Configure VXLAN tunnel with SIP only representing the VTEP.
+- Configure EVPN NVO instance.
+- Configure BGP EVPN address-family and activate IPv4/IPv6 BGP neighbors.
+
+
+The corresponding CONFIG_DB entries are as follows. 
+
+
+```
+VXLAN_TUNNEL_TABLE|{{source_vtep_name}}
+    "src_ip" : {{ipv4_address}}
+    
+VXLAN_EVPN_NVO_TABLE|{{nvo_name}}
+    "source_vtep" : {{source_vtep_name}}
+```
+
+
+
+Following configurations are required for enabling EVPN L3 service on SONiC device:
+
+- Configure L3VNI-VLAN mapping for the given NVO.
+- Configure L3VNI under the corresponding SONiC VRF config
+- Configure L3VNI under the corresponding VRF config in FRR.
+- Configure VRF on VLAN interface associated with the VLAN.
+
+
+The corresponding CONFIG_DB entries are as follows. 
+```
+VLAN|Vlan{{vlan_id}}
+    "vlanid" : {{vlan_id}}
+
+VXLAN_TUNNEL_MAP|{{source_vtep_name}}|{{tunnel_map_name}}
+    "vni" : {{vni_id}}
+    "vlan" : {{vlan_id}}
+
+VRF|{{vrf_name}}
+    "l3vni" : {{vni_id}}
+
```

Comment:
Should the field be "vni", instead of "l3vni", as described above?
key = VRF_TABLE:VRF_NAME ;
fallback = "true"/"false"
vni = 1*8DIGIT ; VNI associated with VRF

Reply:
yes. will update the document.